streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSessionStore.java

```diff
@@ -44,14 +45,53 @@
         this.aggSerde = aggSerde;
     }
 
-
+    @Override
     @SuppressWarnings("unchecked")
+    public void init(final ProcessorContext context, final StateStore root) {
+        this.serdes = new StateSerdes<>(bytesStore.name(),
+                keySerde == null ? (Serde<K>) context.keySerde() : keySerde,
+                aggSerde == null ? (Serde<AGG>) context.valueSerde() : aggSerde);
+
+        bytesStore.init(context, root);
+    }
+
+    // this is optimizing the case when this store is already a bytes store, in which we can avoid Bytes.wrap() costs
+    private static class RocksDBSessionBytesStore extends RocksDBSessionStore<Bytes, byte[]> {
+        RocksDBSessionBytesStore(final SegmentedBytesStore inner) {
+            super(inner, Serdes.Bytes(), Serdes.ByteArray());
+        }
+
+        @Override
+        public KeyValueIterator<Windowed<Bytes>, byte[]> findSessions(final Bytes key, final long earliestSessionEndTime, final long latestSessionStartTime) {
+            final KeyValueIterator<Bytes, byte[]> bytesIterator = bytesStore.fetch(key, earliestSessionEndTime, latestSessionStartTime);
+            return SessionStoreIterator.bytesIterator(bytesIterator, serdes);
+        }
+
+        @Override
+        public void remove(final Windowed<Bytes> key) {
+            bytesStore.remove(SessionKeySerde.bytesToBinary(key));
+        }
+
+        @Override
+        public void put(final Windowed<Bytes> sessionKey, final byte[] aggregate) {
+            bytesStore.put(SessionKeySerde.bytesToBinary(sessionKey), aggregate);
+        }
+    }
+
+    static RocksDBSessionStore<Bytes, byte[]> bytesStore(final SegmentedBytesStore inner) {
```

Comment:
In fact this is because the RocksDBStoreSupplier should trigger this function but it does not. Have fixed it.